internal/service/sync/export/usage_export.go

```diff
@@ -0,0 +1,256 @@
+	for {
+		e.logger.Debugw("fetching batch",
+			"offset", offset,
+			"batch_size", batchSize)
+
+		usageData, err := e.featureUsageRepo.GetFeatureUsageForExport(
+			ctx,
+			request.TenantID,
+			request.EnvID,
+			request.StartTime,
+			request.EndTime,
+			batchSize,
+			offset,
+		)
+		if err != nil {
+			return nil, 0, ierr.WithError(err).
+				WithHint("Failed to fetch feature usage data batch").
+				WithReportableDetails(map[string]interface{}{
+					"offset":     offset,
+					"batch_size": batchSize,
+				}).
+				Mark(ierr.ErrDatabase)
+		}
+
+		// If no data returned, we've reached the end
+		if len(usageData) == 0 {
+			break
+		}
+
+		e.logger.Debugw("fetched batch",
+			"offset", offset,
+			"records_in_batch", len(usageData),
+			"total_so_far", totalRecords+len(usageData))
+
+		// Write batch data to CSV
+		for _, usage := range usageData {
+			// Convert properties map to JSON string
+			propertiesJSON, err := json.Marshal(usage.Properties)
+			if err != nil {
+				e.logger.Warnw("failed to marshal properties, skipping",
+					"usage_id", usage.ID,
+					"error", err)
+				propertiesJSON = []byte("{}")
+			}
+
+			row := []string{
+				usage.ID,
+				usage.TenantID,
+				usage.EnvironmentID,
+				usage.ExternalCustomerID,
+				usage.CustomerID,
+				usage.SubscriptionID,
+				usage.SubLineItemID,
+				usage.PriceID,
+				usage.MeterID,
+				usage.FeatureID,
+				usage.EventName,
+				usage.Source,
+				usage.Timestamp.Format(time.RFC3339),
+				usage.IngestedAt.Format(time.RFC3339),
+				fmt.Sprintf("%d", usage.PeriodID),
+				usage.QtyTotal.String(),
+				string(propertiesJSON),
+				usage.UniqueHash,
+			}
+
+			if err := writer.Write(row); err != nil {
+				return nil, 0, ierr.WithError(err).
+					WithHintf("Failed to write CSV row for usage ID: %s", usage.ID).
+					Mark(ierr.ErrInternal)
+			}
+		}
+
+		totalRecords += len(usageData)
+		offset += batchSize
+
+		// If we got fewer records than batch size, we've reached the end
+		if len(usageData) < batchSize {
+			break
+		}
+	}
+
+	// Always flush the writer, even if no data records were found
+	// This ensures we upload a CSV with headers but no data rows
+	writer.Flush()
+	if err := writer.Error(); err != nil {
+		return nil, 0, ierr.WithError(err).
+			WithHint("Failed to flush CSV writer").
+			Mark(ierr.ErrInternal)
+	}
+
+	csvBytes := buf.Bytes()
+
+	if totalRecords == 0 {
+		e.logger.Infow("no feature usage data found for export - will upload empty CSV with headers only",
+			"tenant_id", request.TenantID,
+			"env_id", request.EnvID,
+			"csv_size_bytes", len(csvBytes))
+	} else {
+		e.logger.Infow("completed batched data fetch and CSV conversion",
+			"total_records", totalRecords,
+			"csv_size_bytes", len(csvBytes))
+	}
+
+	return csvBytes, totalRecords, nil
+}
+
+// Export fetches feature usage data, converts to CSV, and uploads to S3
+func (e *UsageExporter) Export(ctx context.Context, request *ExportRequest) (*ExportResponse, error) {
```

Comment:
move most of the logic to the base service export method and onlyc all maybe prepare data step here